tests/core/failover/task.yaml

```diff
@@ -116,13 +116,7 @@ execute: |
 
     # after rollback, we have the original target snap for a while
     # wait until the kernel and core snap revisions are in place
-    while true ; do
-        current=$(snap list | awk "/^${TARGET_SNAP_NAME} / {print(\$3)}")
-        if [ "$current" = "$(cat prevBoot)" ] ; then
-            break
-        fi
-        sleep 1
-    done
+    retry -n 60 --wait 1 sh -c "test $(snap list | awk "/^${TARGET_SNAP_NAME} / {print(\$3)}") = $(cat prevBoot)"
```

Comment:
Hm I think this is wrong, $(snap list | awk ..) will only be evaluated once when expanding the command. The correct version likely looks like this:
```
#shellcheck disable=<somecheck>
TARGET_SNAP_NAME="$TARGET_SNAPNAME"  \
    retry -n 60 --wait 1 sh -c 'test $(snap list | awk "/^${TARGET_SNAP_NAME} / {print(\$3)}") = $(cat prevBoot)'
```

Reply:
@bboozzoo you are right, thanks for the review. Fix in progress

Reply:
```
 TARGET_SNAP_NAME="$TARGET_SNAP_NAME" retry -n 60 --wait 1 sh -c 'test $(snap list | awk "/^${TARGET_SNAP_NAME} / {print(\$3)}") = $(cat prevBoot)'
                                                                ^-- SC2016: Expressions don't expand in single quotes, use double quotes for that.
```
we need to explicitly disable SC2016